File: src/main/resources/assets/spark_core/js/blueprint_engine/data/BlueprintGraph.js
// BlueprintGraph.js
// 蓝图图结构定义

/**
 * @typedef {Object} BlueprintGraph
 * @property {string} id - 唯一标识
 * @property {string} name - 图名称
 * @property {BlueprintNode[]} nodes - 节点数组
 * @property {BlueprintConnection[]} connections - 连接数组
 * @property {BlueprintVariable[]} variables - 变量数组
 * @property {Object} metadata - 附加元数据
 */

export class BlueprintGraph {
  /**
   * @param {Object} opts
   * @param {string} opts.id
   * @param {string} opts.name
   * @param {BlueprintNode[]} opts.nodes
   * @param {BlueprintConnection[]} opts.connections
   * @param {BlueprintVariable[]} opts.variables
   * @param {Object} opts.metadata
   */
  constructor({ id = '', name = '', nodes = [], connections = [], variables = [], metadata = {} } = {}) {
    this.id = id;
    this.name = name;
    this.nodes = nodes;
    this.connections = connections;
    this.variables = variables;
    this.metadata = metadata;
  }
}

File: src/main/kotlin/cn/solarmoon/spark_core/blueprint/data/BlueprintGraph.kt
package cn.solarmoon.spark_core.blueprint.data

import java.util.*

/**
 * BlueprintGraph 数据类
 * @property id 唯一标识
 * @property name 图名称
 * @property nodes 节点列表
 * @property connections 连接列表
 * @property variables 变量列表
 * @property metadata 附加元数据
 */
data class BlueprintGraph(
    val id: String = UUID.randomUUID().toString(),
    val name: String = "",
    val nodes: MutableList<BlueprintNode> = mutableListOf(),
    val connections: MutableList<BlueprintConnection> = mutableListOf(),
    val variables: MutableList<BlueprintVariable> = mutableListOf(),
    val metadata: Map<String, Any?> = emptyMap()
)
